src/rpc/mempool.cpp

```diff
@@ -376,6 +376,90 @@ RPCHelpMan getmempoolentry()
+{
+    return RPCHelpMan{"getmempooltxspendingprevout",
+                "Scans the mempool to find transactions spending any of the given outputs",
+                {
+                    {"outputs", RPCArg::Type::ARR, RPCArg::Optional::NO, "The transaction outputs that we want to check, and within each, the txid (string) vout (numeric).",
+                        {
+                            {"", RPCArg::Type::OBJ, RPCArg::Optional::OMITTED, "",
+                                {
+                                    {"txid", RPCArg::Type::STR_HEX, RPCArg::Optional::NO, "The transaction id"},
+                                    {"vout", RPCArg::Type::NUM, RPCArg::Optional::NO, "The output number"},
+                                },
+                            },
+                        },
+                    },
+                },
+                RPCResult{
+                    RPCResult::Type::ARR, "", "",
+                    {
+                        {RPCResult::Type::OBJ, "", "",
+                        {
+                            {RPCResult::Type::STR_HEX, "txid", "the transaction id of the spent output"},
+                            {RPCResult::Type::NUM, "vout", "the vout value of the spent output"},
+                            {RPCResult::Type::STR_HEX, "spendingtxid", "the transaction id of the mempool transaction spending this output (omitted if unspent)"},
+                        }},
+                    }
+                },
+                RPCExamples{
+                    HelpExampleCli("getmempooltxspendingprevout", "\"[{\\\"txid\\\":\\\"a08e6907dbbd3d809776dbfc5d82e371b764ed838b5655e72f463568df1aadf0\\\",\\\"vout\\\":3}]\"")
+                    + HelpExampleRpc("getmempooltxspendingprevout", "\"[{\\\"txid\\\":\\\"a08e6907dbbd3d809776dbfc5d82e371b764ed838b5655e72f463568df1aadf0\\\",\\\"vout\\\":3}]\"")
+                },
+        [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
+{
+    RPCTypeCheckArgument(request.params[0], UniValue::VARR);
+    const UniValue& output_params = request.params[0];
+    if (output_params.empty()) {
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid parameter, outputs are missing");
+    }
+
+    std::vector<COutPoint> outputs;
+    outputs.reserve(output_params.size());
+
+    for (unsigned int idx = 0; idx < output_params.size(); idx++) {
+        const UniValue& o = output_params[idx].get_obj();
+
+        RPCTypeCheckObj(o,
+            {
+                {"txid", UniValueType(UniValue::VSTR)},
+                {"vout", UniValueType(UniValue::VNUM)},
+            });
+
+        const uint256 txid(ParseHashO(o, "txid"));
+        const int nOutput = find_value(o, "vout").get_int();
+        if (nOutput < 0) {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid parameter, vout cannot be negative");
+        }
+
+        const COutPoint output(txid, nOutput);
+        outputs.push_back(output);
```

Comment:
```suggestion
        outputs.emplace_back(txid, nOutput);
```

Change this can avoid a copy of the object

Reply:
Thanks, done in https://github.com/bitcoin/bitcoin/pull/24408/commits/8e6db71ff6cf2c8f724183996b4842fcf53d16b0

Reply:
The build started failing after this change, so I'm reverting it.
To be honest, I don't think we care about such optimizations on just a few bytes here.

Reply:
>The tests started failing after this change, so I'm reverting it.

Mh! strange the Ci looks like failing again.

>To be honest, I don't think we care about such optimizations on just a few bytes here.

Right, this can be also a followup PR too

Reply:
Test failure is due to stricter documentation checking in current master, not due to commit 8e6db71ff6cf2c8f724183996b4842fcf53d16b0 . I left a comment with a potential fix.